src/pip/_internal/cache.py

```diff
@@ -102,13 +103,18 @@ def _link_for_candidate(self, link, candidate):
 
         return index.Link(path_to_url(path))
 
+    def cleanup(self):
```

Comment:
I should note that @xoviat had proposed in #4714 to make caches context managers, to avoid using this `cleanup()` pattern.

I'm still confused about how I feel about that. FWIW, we can do that in a later PR. I have some other ideas too but I'd prefer to experiment with that after this PR is done.

Reply:
@rhettinger What do you think about using context managers for cleanup? You're the person that I got this from.

Reply:
Just to be clear, I definitely think using a context manager is a good idea. But I agree with @pradyunsg that it's something we should defer until a followup PR. Let's get the functionality landed then address the cleanup.

Reply:
I agree with that; I was just wondering what others thought in principle.